src/providers/ldap/sdap.c

```diff
@@ -1614,10 +1614,14 @@ sdap_get_primary_fqdn(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    primary_fqdn = sss_create_internal_fqname(tmp_ctx, shortname, dom->name);
-    if (primary_fqdn == NULL) {
-        ret = ENOMEM;
-        goto done;
+    if (strchr(shortname, '@') == NULL) {
```

Comment:
I removed this patch and modified the caller rfc2307bis_nested_groups_step() to check if sdap_get_group_primary_name() must be called or not.